draco/synthesis/noise.py

```diff
@@ -331,7 +390,7 @@ def draw_complex_wishart(C, n):
 
 @contextlib.contextmanager
 def mpi_random_seed(seed, extra=0):
```

Comment:
You could maybe just merge these two, and set/reset both seeds in the same context manager. You could also add some parameters to control it, i.e. `def mpi_random_seed(seed, extra=0, set_numpy=True, set_randomgen=True)`

Reply:
mpi_randomgen needs to be passed the generator, as well as the seed, but I can check for the presence of a generator. If present, we can set/reset its seed.

Reply:
Also, a bit complicated by the yield statement. randomgen needs to yield its generator back.

So I can merge them, but the decision of which they set/reset will likely be exclusive.